backend/inspirehep/records/api/mixins.py

```diff
@@ -13,7 +14,94 @@
 LOGGER = structlog.getLogger()
 
 
-class CitationMixin:
+class AuthorsPapersMapExtension:
+    def generate_entries_for_authors_in_authors_records_table(self):
+        """Generates AuthorsRecords objects table based on record data for authors"""
+
+        authors_ids_field = "authors.ids"
+        table_entries_buffer = []
+        for author in self.get_value(authors_ids_field, []):
+            if author:
+                for id in author:
+                    try:
+                        table_entries_buffer.append(
+                            AuthorsRecords(
+                                author_id=id["value"],
+                                id_type=id["schema"],
+                                record_id=self.id,
+                            )
+                        )
+                    except KeyError:
+                        LOGGER.exception(
+                            "id entry is missing keys",
+                            recid=self.get("control_number"),
+                            uuid=str(self.id),
+                        )
+        return table_entries_buffer
+
+    def generate_entries_for_conferences_in_authors_records_table(self):
+        """Generates AuthorsRecords objects table based on record data for collaborations"""
+        collaborations_field = "collaborations.value"
+        table_entries_buffer = []
+        for collaboration in self.get_value(collaborations_field, []):
+            table_entries_buffer.append(
+                AuthorsRecords(
+                    author_id=collaboration, id_type="collaboration", record_id=self.id
+                )
+            )
+        return table_entries_buffer
+
+    def update_authors_records_table(self):
+        """Puts all authors ids and collaborations in authors_records table"""
+        deleted_count = self.delete_authors_records_table_entries()
+
+        # check if authors and conferencesshould be added to the table
+        if (
+            self.get("deleted", False)
+            or self.pid_type not in ["lit"]
+            or "Literature" not in self["_collections"]
```

Comment:
here, for example, you are already making it specific to literature

Reply:
Currently Our citations are limited the same way. https://github.com/inspirehep/inspirehep/blob/66f310daacb68f854cf22bcbe48b1a46e4e92259/backend/inspirehep/records/api/mixins.py#L78-L80

Reply:
I thought for citations we were limiting like that because data can be cited but can't cite. Anyhow, I don't think it is important for it to work now on data, just wanted to point that out